server/sonar-server/src/main/java/org/sonar/server/ui/ws/ComponentNavigationAction.java

```diff
@@ -164,6 +169,19 @@ private void writeProfiles(JsonWriter json, DbSession session, ComponentDto comp
     json.endArray();
   }
 
+  private void writeQualityGate(JsonWriter json, DbSession session, ComponentDto component) {
+    Optional<QualityGateFinder.QualityGateData> qualityGateData = qualityGateFinder.getQualityGate(session, component.getId());
+    if (!qualityGateData.isPresent()) {
+      return;
+    }
+    QualityGateDto qualityGateDto = qualityGateData.get().getQualityGate();
+    json.name("qualityGate").beginObject()
+      .prop("key", qualityGateDto.getId())
```

Comment:
No choice, I would have loved to changed it, but here my scope is not to rewrite existing WS, but I've created to update all api/qualitygates WS => https://jira.sonarsource.com/browse/SONAR-8507